federation/apis/federation/types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// ClusterSpec describes the attributes of a kubernetes cluster.
+type ClusterSpec struct {
+	// A map of client CIDR to server address.
+	// This is to help clients reach servers in the most network-efficient way possible.
+	// Clients can use the appropriate server address as per the CIDR that they match.
+	// In case of multiple matches, clients should use the longest matching CIDR.
+	ServerAddressByClientCIDRs []unversioned.ServerAddressByClientCIDR `json:"serverAddressByClientCIDRs" patchStrategy:"merge" patchMergeKey:"clientCIDR"`
+	// the type (e.g. bearer token, client certificate etc) and data of the credential used to access cluster.
+	// It’s used for system routines (not behalf of users)
+	Credential string `json:"credential,omitempty"`
```

Comment:
Just to reiterate, I dont think that this will work.
We will need to add specific fields as I explained in https://github.com/kubernetes/kubernetes/pull/23847#discussion_r59301275.
But fine to get this in as is and fix later.